### PR TITLE
Add docs on how to run script within Docker container

### PR DESCRIPTION
**Pull request by juanjoqmelian**
Found a bit confusing how to run Dependabot inside of a Docker container initially, so trying to make it clear with this PR.
One of the misunderstandings was the `Running with Docker` section in dependabot-core and the `docker-dev-shell` script. Not sure if other people are having the same iasue, but initially I didn't realise that the script was meant to be used to make changes on dependabot-core for developers and not to run dependabot against a git repo itself.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,6 +57,44 @@ Build the helpers you want to use (you'll also need the corresponding language i
...
+```shell
+$ docker pull dependabot/dependabot-core
+$ docker run -ti -w /home/dependabot/dependabot-script --env-file env.list -v "$(pwd):/home/dependabot/dependabot-script" dependabot/dependabot-core
+
+(Now you'd be inside the Docker container)
+/home/dependabot/dependabot-script# bundle install
+/home/dependabot/dependabot-script# ./generic-update-script.rb (Add '#!/usr/bin/env ruby' to execute the script from the shell)
+```
+
+Please notice that you'll have to create an `env.list` file with your environment properties. For example:
```
Comment (LeoColomb) [inline README.md]: Using file is a good idea, but I think passing env via the command line arguments is better to begin. You can always link to the official docker documentation for advanced users.

File: README.md [hunk 2]
```diff
@@ -57,6 +57,44 @@ Build the helpers you want to use (you'll also need the corresponding language i
...
+
+```shell
+GITHUB_ENTERPRISE_ACCESS_TOKEN=xxxxxxxxx
+GITHUB_ENTERPRISE_HOSTNAME=mycompany.github.com
```
Comment (LeoColomb) [inline README.md]: You need to explain the case for the four sources: GitHub, GHE, GitLab & Azure DevOps.

File: README.md [hunk 3]
```diff
@@ -57,6 +57,44 @@ Build the helpers you want to use (you'll also need the corresponding language i
...
+In order to do that, you'll have to pull the image from Docker Hub and mount your working directory into the container.
+You'll also have to set several environment variables to make the script work with your configuration, 
+as specified in the documentation.
+
+Steps:
+```shell
+$ docker pull dependabot/dependabot-core
+$ docker run -ti -w /home/dependabot/dependabot-script --env-file env.list -v "$(pwd):/home/dependabot/dependabot-script" dependabot/dependabot-core
+
+(Now you'd be inside the Docker container)
```
Comment (LeoColomb) [inline README.md]: Prefer running command outside of the container. Entering the container is considered as advanced usage. 

This also let you specify that the install step is required only once. 
For example:

1. Pull the Docker image
```shell
docker pull dependabot/dependabot-core
```

2. Install dependencies
```shell
docker run --rm dependabot/dependabot-core bundle install -j 3 --path vendor
```

3. Run dependabot
```shell
docker run --rm dependabot/dependabot-core -v ./:working-dir -e PROJECT=foo/bar bundle exec ruby ./script
```

Comment (juanjoqmelian) [reply README.md]: I've addressed your comments @LeoColomb.
Let me know if it looks ok to you.

**Comment by greysteil** [comment]
Thanks team!

**Pull request merged**